.github/workflows/CI.yml

```diff
@@ -16,9 +16,14 @@ jobs:
           node-version: 12.x
       - name: Environment Information
         run: npx envinfo
-      - name: build it
+      - name: Build and Test on ubuntu and macOS
+        if: matrix.os != 'windows-latest'
         run: |
           cmake .
           cmake --build . --target check
-        env:
-          CI: true
+      - name: Build and Test on windows
+        if: matrix.os == 'windows-latest'
+        run: |
+          "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\Build\vcvars64.bat"
+          cmake -G "CodeBlocks - NMake Makefiles" .
+          cmake --build . --target check
```

Comment:
Since `cmake --build . --target check` is common in both of these, maybe we should have a step for building and a separate shared step for running `check`.